### PR TITLE
Add template gigaset linekey support

### PR DESCRIPTION
**Pull request by alepolidori**
- I've added a feature to make provisioning unlock for all linekey as explained [here](https://teamwork.gigaset.com/gigawiki/display/GPPPO/FAQ+Maxwell+-+Auto+provisioning+unlock+function+keys)
- I've added a feature to always show the display name for Gigaset line keys
- I've added a feature to always show the icons for Gigaset line keys
- I've added a feature to always show the key indicator as bold for Gigaset line keys

### CODE REVIEW

File: data/templates/gigaset.macros [hunk 1]
```diff
@@ -1,3 +1,36 @@
+#
+# linekey_type_map -- translate "linekey_type_"
+#
+{% macro linekey_type_map(linekey_type, number, linekey_value, dndtoggle, pickup_group) %}
+{% if linekey_type == "blf" %}
+    {{- '<param name="PhoneUI.Keys.FunctionKeys.' -}} {{- number -}} {{- '.Type" value="4" />' }}
```
Comment (DavidePrincipi) [inline data/templates/gigaset.macros]: Please use a simpler notation, like:

```suggestion
    <param name="PhoneUI.Keys.FunctionKeys.{{ number }}.Type" value="4" />
```

File: data/templates/gigaset.macros [hunk 2]
```diff
@@ -1,3 +1,36 @@
...
+    {{ '<param name="PhoneUI.Keys.FunctionKeys.' -}} {{- number -}} {{- '.PhoneNumber" value="' -}} {{ linekey_value }} {{- '" />' }}
```
Comment (DavidePrincipi) [inline data/templates/gigaset.macros]: Other example...

```suggestion
    <param name="PhoneUI.Keys.FunctionKeys.{{ number }}.PhoneNumber" value="{{ linekey_value }}"  />
```

File: data/templates/gigaset-Maxwell.tmpl [hunk 3]
```diff
@@ -51,7 +51,24 @@
...
+    <param name="Telephony.ShowKeysDisplayName" value="1"/>
+    <param name="PhoneUI.Keys.FunctionKeys.Icons" value="1"/>
+    <param name="PhoneUI.Keys.FunctionKeys.Indicator" value="1"/>
   </nvm>
+  <custom>
+    <step type="UnlockAllFunctionKeys" />
+  </custom>
+  {# end line keys #}
 </provisioning>
 {% endautoescape %}
```
[the remark this answers is not in the document]
Comment (alepolidori) [reply data/templates/gigaset-Maxwell.tmpl]: It was not present before this PR: is it strictly necessary for phone to work correctly or it's a our convention ?

[the remark this answers is not in the document]
Comment (DavidePrincipi) [reply data/templates/gigaset-Maxwell.tmpl]: So it's my guilty!
If there is no reason to omit it, there must be a sane EOL at EOF.

[the remark this answers is not in the document]
Comment (alepolidori) [reply data/templates/gigaset-Maxwell.tmpl]: done

**Pull request merged**